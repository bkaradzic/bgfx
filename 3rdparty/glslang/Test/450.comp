#version 450 core
