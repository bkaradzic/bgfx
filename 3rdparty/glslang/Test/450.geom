#version 450 core

in gl_PerVertex {
    float gl_CullDistance[3];
} gl_in[];

out gl_PerVertex {
    float gl_CullDistance[3];
};

void main()
{
    gl_CullDistance[2] = gl_in[1].gl_CullDistance[2];
}
