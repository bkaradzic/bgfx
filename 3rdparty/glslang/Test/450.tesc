#version 450 core

in gl_PerVertex {
    float gl_CullDistance[3];
} gl_in[gl_MaxPatchVertices];

out gl_PerVertex {
    float gl_CullDistance[3];
} gl_out[4];

void main()
{
    gl_out[gl_InvocationID].gl_CullDistance[2] = gl_in[1].gl_CullDistance[2];
}
