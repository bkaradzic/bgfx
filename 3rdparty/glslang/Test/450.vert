#version 450 core

out gl_PerVertex {
    float gl_CullDistance[3];
};

void main()
{
    gl_CullDistance[2] = 4.5;
}

out bool outb;         // ERROR
out sampler2D outo;    // ERROR
out float outa[4];
out float outaa[4][2];
struct S { float f; };
out S outs;
out S[4] outasa;
out S outsa[4];
struct SA { float f[4]; };
out SA outSA;
struct SS { float f; S s; };
out SS outSS;
