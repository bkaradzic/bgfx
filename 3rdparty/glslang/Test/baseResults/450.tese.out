450.tese
Warning, version 450 is not yet complete; most version-specific features are present, but some are missing.

Shader version: 450
input primitive = none
vertex spacing = none
triangle order = none
0:? Sequence
0:11  Function Definition: main( (global void)
0:11    Function Parameters: 
0:13    Sequence
0:13      move second child to first child (temp float)
0:13        direct index (temp float CullDistance)
0:13          gl_CullDistance: direct index for structure (out 3-element array of float CullDistance)
0:13            'anon@0' (out block{out 3-element array of float CullDistance gl_CullDistance})
0:13            Constant:
0:13              3 (const uint)
0:13          Constant:
0:13            2 (const int)
0:13        direct index (temp float CullDistance)
0:13          gl_CullDistance: direct index for structure (in 3-element array of float CullDistance)
0:13            direct index (temp block{in 3-element array of float CullDistance gl_CullDistance})
0:13              'gl_in' (in 32-element array of block{in 3-element array of float CullDistance gl_CullDistance})
0:13              Constant:
0:13                1 (const int)
0:13            Constant:
0:13              0 (const int)
0:13          Constant:
0:13            2 (const int)
0:?   Linker Objects
0:?     'gl_in' (in 32-element array of block{in 3-element array of float CullDistance gl_CullDistance})
0:?     'anon@0' (out block{out 3-element array of float CullDistance gl_CullDistance})


Linked tessellation evaluation stage:

ERROR: Linking tessellation evaluation stage: At least one shader must specify an input layout primitive

Shader version: 450
input primitive = none
vertex spacing = equal_spacing
triangle order = ccw
0:? Sequence
0:11  Function Definition: main( (global void)
0:11    Function Parameters: 
0:13    Sequence
0:13      move second child to first child (temp float)
0:13        direct index (temp float CullDistance)
0:13          gl_CullDistance: direct index for structure (out 3-element array of float CullDistance)
0:13            'anon@0' (out block{out 3-element array of float CullDistance gl_CullDistance})
0:13            Constant:
0:13              3 (const uint)
0:13          Constant:
0:13            2 (const int)
0:13        direct index (temp float CullDistance)
0:13          gl_CullDistance: direct index for structure (in 3-element array of float CullDistance)
0:13            direct index (temp block{in 3-element array of float CullDistance gl_CullDistance})
0:13              'gl_in' (in 32-element array of block{in 3-element array of float CullDistance gl_CullDistance})
0:13              Constant:
0:13                1 (const int)
0:13            Constant:
0:13              0 (const int)
0:13          Constant:
0:13            2 (const int)
0:?   Linker Objects
0:?     'gl_in' (in 32-element array of block{in 3-element array of float CullDistance gl_CullDistance})
0:?     'anon@0' (out block{out 3-element array of float CullDistance gl_CullDistance})

